test/integration/udpingress_test.go

```diff
@@ -151,6 +164,217 @@ func TestUDPIngressEssentials(t *testing.T) {
+	deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, configVolume)
+
+	t.Log("deploying coredns")
+	deployment, err = env.Cluster().Client().AppsV1().Deployments(ns.Name).Create(ctx, deployment, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up deployment %s", deployment.Name)
+		assert.NoError(t, env.Cluster().Client().AppsV1().Deployments(ns.Name).Delete(ctx, deployment.Name, metav1.DeleteOptions{}))
+	}()
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeNodePort)
+	service, err = env.Cluster().Client().CoreV1().Services(ns.Name).Create(ctx, service, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up the service %s", service.Name)
+		assert.NoError(t, env.Cluster().Client().CoreV1().Services(ns.Name).Delete(ctx, service.Name, metav1.DeleteOptions{}))
+	}()
+
+	// Test initial configuration with UDP only
+	t.Log("exposing DNS service via UDPIngress")
+	udp := &kongv1beta1.UDPIngress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "minudp",
+			Annotations: map[string]string{
+				annotations.IngressClassKey: ingressClass,
+			},
+		},
+		Spec: kongv1beta1.UDPIngressSpec{Rules: []kongv1beta1.UDPIngressRule{
+			{
+				Port: 9999,
+				Backend: v1beta1.IngressBackend{
+					ServiceName: service.Name,
+					ServicePort: int(service.Spec.Ports[0].Port),
+				},
+			},
+		}},
+	}
+	c, err := clientset.NewForConfig(env.Cluster().Config())
+	assert.NoError(t, err)
+	udp, err = c.ConfigurationV1beta1().UDPIngresses(ns.Name).Create(ctx, udp, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("ensuring UDPIngress %s is cleaned up", udp.Name)
+		if err := c.ConfigurationV1beta1().UDPIngresses(ns.Name).Delete(ctx, udp.Name, metav1.DeleteOptions{}); err != nil {
+			if !errors.IsNotFound(err) {
+				require.NoError(t, err)
+			}
+		}
+	}()
+
+	t.Log("configurating a dns query and clients")
+	query := new(dns.Msg)
+	query.Id = dns.Id()
+	query.Question = make([]dns.Question, 1)
+	query.Question[0] = dns.Question{Name: "kernel.org.", Qtype: dns.TypeA, Qclass: dns.ClassINET}
+	dnsUDPClient := new(dns.Client)
+	dnsTCPClient := dns.Client{Net: "tcp"}
+
+	t.Logf("checking udpingress %s status readiness.", udp.Name)
+	ingCli := c.ConfigurationV1beta1().UDPIngresses(ns.Name)
+	assert.Eventually(t, func() bool {
+		curIng, err := ingCli.Get(ctx, udp.Name, metav1.GetOptions{})
+		if err != nil || curIng == nil {
+			return false
+		}
+		ingresses := curIng.Status.LoadBalancer.Ingress
+		for _, ingress := range ingresses {
+			if len(ingress.Hostname) > 0 || len(ingress.IP) > 0 {
+				t.Logf("udpingress hostname %s or ip %s is ready to redirect traffic.", ingress.Hostname, ingress.IP)
+				return true
+			}
+		}
+		return false
+	}, 120*time.Second, 1*time.Second, true)
```

Comment:
I think this was probably a leftover from the UDPIngress tests, but maybe since we're here should we reduce this to 30s? I was likely the original culprit of these timings, but since we're here in retrospect 2m seems incredibly long.

Reply:
86f0dfca7ad9c67b6237e428a6553467ed261b25